src/map/lua/lua_loot.cpp

```diff
@@ -69,7 +65,47 @@ void CLuaLootContainer::addItem(uint16 item, uint16 rate, sol::variadic_args va)
 
 void CLuaLootContainer::addGroup(uint16 groupRate, sol::table items)
 {
-    DropGroup_t group(RATE_PERCENTAGES[groupRate]);
+    addGroupToContainer(RATE_PERCENTAGES[groupRate], items, false);
+}
+
+/************************************************************************
+ *  Function: addItemFixed()
+ *  Purpose : Adds an item to the loot container with a fixed drop rate
+ *  Example : loot:addItemFixed(xi.items.ANCIENT_BEASTCOIN, 500, 2);
+ *  Notes   : Last parameter, quantity, is optional and defaults to 1
+ *            Fixed drop rate is 0-1000.
+ ************************************************************************/
+
+void CLuaLootContainer::addItemFixed(uint16 item, uint16 rate, sol::variadic_args va)
+{
+    addItemToContainer(item, rate, va, true);
+}
+
+/************************************************************************
+ *  Function: addGroupFixed()
+ *  Purpose : Adds a group of items to the loot container with a fixed drop rate
+ *  Example : loot:addGroup(500, { { item = xi.items.ANCIENT_BEASTCOIN, weight = 100 } });
+ *  Notes   : Item table is a list of tables with "item" key and an optional "weight" which defaults to 1
+ *            Fixed drop rate is 0-1000.
+ ************************************************************************/
+
+void CLuaLootContainer::addGroupFixed(uint16 groupRate, sol::table items)
+{
+    addGroupToContainer(groupRate, items, true);
+}
+
+void CLuaLootContainer::addItemToContainer(uint16 item, uint16 rate, sol::variadic_args va, bool hasFixedRate)
+{
+    const auto quantity = va.get_type(0) == sol::type::number ? va.get<uint16>(0) : 1;
+    for (uint16 i = 0; i < quantity; ++i)
```

Comment:
Should this limit to the max size of a treasure pool? Or is that handled elsewhere? I suppose most of the time that's a moot point because mobs never have enough drops the flood the pool from one mob anyway

Reply:
That'll be handled when the drops are calculated. Mob drop lists come from SQL, `ITEM_DROPS` can temporarily add to the drop list, and then the drops are calculated.

Reply:
@WinterSolstice8 I'd say no, there are certain cases where it would but the handling for those situations lies elsewhere and we'd a hard limit in treasure pool for our current drops, which I can actually make a reasonable argument is actually not retail in the 1st place (if it overflows and items are dropped, thats not necessarily wrong, tho in 99% of cases retail uses groupings to avoid that causing loss of rare loot)